src/components/Detail/Detail.js

```diff
@@ -1,9 +1,17 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { Link } from 'react-router-dom';
+import { fetchFilm } from '../../store/reducer';
 import Logo from '../Logo/Logo';
 import classes from './Detail.css';
 
 class Detail extends Component {
+  componentDidMount() {
+    const { id } = this.props.match.params;
+    this.props.fetchFilm(id);
```

Comment:
add the same check and fetch in `componentDidUpdate`. By adding it you will no longer need to use your action `selectFilmOnClick` inside your `Films` component